admin-dev/themes/new-theme/js/components/generatable-input.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * 2007-2019 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+/**
+ * Generates random values for inputs.
+ *
+ * Usage:
+ *
+ * There should be a button in HTML with 2 required data-* properties:
+ *    1. data-target-input-id - input id for which value should be generated
+ *    2. data-generated-value-size -
+ *
+ * Example button: <button class="js-generator-btn"
+ *                         data-target-input-id="my-input-id"
+ *                         data-generated-value-length="16"
+ *                 >
+ *                     Generate!
+ *                 </button>
+ *
+ * In JavaScript you have to enable this functionality using GeneratableInput component like so:
+ *
+ * const generateableInput = new GeneratableInput();
+ * generateableInput.attachOn('.js-generator-btn'); // every time our button is clicked
+ *                                                  // it will generate random value of 16 characters
+ *                                                  // for input with id of "my-input-id"
+ *
+ * You can attach as many different buttons as you like using "attachOn()" function
+ * as long as 2 required data-* attributes are present at each button.
+ */
+export default class GeneratableInput {
```

Comment:
Just an idea: since these JS extensions we have created for the BO new theme are quite modular and uncoupled, we could easily (OK, maybe not easily but not difficult either) test each of theme individually.

This might requires something like http://phantomjs.org/api/webpage/method/set-content.html
I'm thinking about:
- we write the sample HTML needed, for example (1)
- we write the JS sample needed to include the extension
- we perform the required action, here a click on "generate input !"
- we check the DOM was updated in the right way
(1):
```
<button class="js-generator-btn"
     data-target-input-id="my-input-id"
     data-generated-value-length="16"
>
 Generate!
</button>
```

Wdyt ?

Reply:
i've never done that, but it sounds like a great idea. :bulb:

Reply:
After having a look at it, it seems this can become quite an heavy topic.
Moreover, running such tests is one thing, getting them to be run by CI is another (javascript dependencies to install, webpack compilation ...)

So another PR 😉